tools/js-sdk-release-tools/src/common/utils.ts

```diff
@@ -422,16 +422,43 @@ export async function cleanUpPackageDirectory(
     packageDirectory: string,
     runMode: RunMode,
 ): Promise<void> {
-    // Preserve test directory and assets.json file only in Release and Local modes
-    // In SpecPullRequest and Batch modes, remove everything
-    const pipelineRunMode = runMode !== RunMode.SpecPullRequest && runMode !== RunMode.Batch;
-    const entriesToPreserveForMgmtPackages = pipelineRunMode ? ["test", "assets.json"] : [];
+    // Check if directory exists first
+    if (!fs.existsSync(packageDirectory)) {
+        logger.info(`Directory ${packageDirectory} doesn't exist yet, nothing to clean up.`);
+        return;
+    }
 
     const modularSDKType = getModularSDKType(packageDirectory);
-    if (modularSDKType === ModularSDKType.DataPlane && pipelineRunMode) {
-        return;
+    const pipelineRunMode = runMode !== RunMode.SpecPullRequest && runMode !== RunMode.Batch;
+
+    if (modularSDKType === ModularSDKType.DataPlane) {
```

Comment:
here should be `getSDKType(packageDirectory) === SDKType.RestLevelClient || ModularSDKType.DataPlane` ?